### ISSUE TITLE
The Community Mind - Using Questions to Inspire Innovation

### ISSUE THREAD

**Issue by wip-abramson** [Blockternship/projects#5]
Hi, my name is Will Abramson. I'm a recent Computer Science graduate and Software Developer and have been exploring the blockchain space for about 8 months. Hoping to integrate blockchain technology into a side project that I have been working on since university. 

## Overview

[The Community Mind](https://github.com/wip-abramson/TheCommunityMind) aims to become a place to capture peoples open "beautiful" questions. I was inspired by a book called [A More Beautiful QUestion](http://amorebeautifulquestion.com/). I envision the website will act like a cross between twitter and medium but only for questions, no answers. I want to build a site that will help people think for themselves by creating a web of unanswered open questions that anyone can both contribute to and explore for inspiration.

If you need some convincing why questions are at the root of creativity check out this [paper](https://umanitoba.ca/admin/human_resources/change/media/the-art-of-powerful-questions.pdf) and if you are really interested I recommend reading the book.

The project is a work in progress but It is reasonably well developed already. I would love some help to really make this as good as I believe it could be.

## Skills
I am an experienced Software Developer in both Java and JavaScript. The project is currently build using React with a Node backend. I also use GraphQl and Apollo. I have a solid understanding of blockchain fundamentals (I think) and have developed in Ethereum before using tools like Truffle and Ganache although not on a major project. Interested in trying out and learning new tools. Looking to collaborate and learn. Happy to help others.

## Blockchain Ideas
I have a couple of ideas for how I believe utilising blockchain could enhance this application.

- I would like to use blockchain as a way to rank questions. Particularly interested in looking into Token Curated Registries, see [here](https://blog.oceanprotocol.com/the-layered-tcr-56cc5b4cdc45) and [here](https://medium.com/@simondlr/tokens-2-0-curved-token-bonding-in-curation-markets-1764a2e0bee5). I want a way to incentivise and curate the best questions and questioners and think blockchain presents a potential solution. I have already integrated [Simple Token](https://ost.com/) as a PoC using it to create a micro-economy around asking questions. Here is my final video [submission](https://www.youtube.com/watch?v=kMABXKGoiAk). Whether I stick with this or look into other ideas I am not sure. Open to ideas.
- Also particularly interested in looking into Self-Sovereign Identity solutions. Looking to integrate one, possibly [uPort](https://www.uport.me/) into the Community Mind. Interested in looking at how users can save the questions they ask and potentially access them in other applications that develop to support The Community Mind - One day. Interested in data interoperability and empowering users with the data that they create.

## Philosophy
This project has been open source from the start and I intend to keep it that way. I want this to be a tool to encourage open collaboration from everyone on ideas, whilst at the same time removing answers will encourage independent thinking. I want to open source creativity!

## Next Steps
The most important area I want to focus on is content discovery. How can I incentivise users to contribute and curate questions in the Community Mind?

Basically, there is a lot to do, but I really believe in this project and its potential. Would welcome any and all contributors and advice. 

This is the currently deployed version of [The Community Mind](thecommunitymind.com).

GitHub: @wip-abramson
KeyBase:@wip-abramson
Twitter: [@Wip_ID](https://twitter.com/Wip_ID?lang=en)
Medium:  [@wip-abramson](https://medium.com/@wip.abramson)

**Comment by YalorMewn** [comment]
@wip-abramson this looks like a **really** amazing project, I thank you for giving such a well detailed and thoughtful submission. Please give us some time to review the materials you have provided and share this with all the teams before we start asking some questions, stay tuned for more 💯 🗣

**Comment by Bijan-Massoumi** [comment]
I'm down to help, but what benefit does a Blockchain add for curation that a server doesn't?

**Comment by wip-abramson** [comment]
I hope it would mean the more people are interested in topics and ask good questions around them,  the more influence they could assert over that topic. I would like the network of questioners to decide which questions deserve the most attention from users as well as being responsible for preventing bad actors by rejecting there submissions.

It would basically help create a decentralised community around asking questions with the best questions and questioners rising to the top as determined by the community.

Blockchain can provide interesting solutions to a number of questions such as:

- How do we prevent useless questions and spam entering the site?
- Who should decide which topics users question about? Can anyone add topics? Which are the most popular topics? How can we focus topics instead of having multiple similarly names topics?
- Which questions most deserve a users time?

As the application I don't want to make those decisions for the community that I hope will one day be creating these questions.

**Comment by YalorMewn** [comment]
@wip-abramson this is very much in line with the creative spirit of @Blockternship/giveth and the our #Blockchain4Good movement and I look forward to seeing more !  So welcome to the Blockternship, can we fork an existing repo in for you or would you like to open a new one ?

**Comment by wip-abramson** [comment]
@YalorMewn great! You can fork in my existing repo - https://github.com/wip-abramson/TheCommunityMind. However it currently is linked with the OST POC that I took part in meaning that you need some secret API keys to run it locally and access the OST server. 

I haven't yet decided if I should stick with OST or look at other options. It would be great if I could talk this over with somebody as part of the blockternship!

Not sure if its best to comment out the OST code for now or just let people know in the Readme that should ask me for the keys. I am new to handling secret env variables so if anyone has any suggestions that would be great.

**Comment by YalorMewn** [comment]
Hey @quazia @deamme do you have any thoughts on this ?

**Comment by YalorMewn** [comment]
Hey @wip-abramson I would like to invite you to the first call of the hackathon, would you send me a quick email here YalorTackson@gmail.com also you can add yourself to the event [here](https://calendar.google.com/event?action=TEMPLATE&tmeid=NjlxM2dtdDd2bHA5ODNxbzNsYm5vOTMyZDkgeWFsb3J0YWNrc29uQG0&tmsrc=yalortackson%40gmail.com)

**Comment by Quazia** [comment]
@wip-abramson I mean, for the most part you'll be able to move to move your state storage over to blockchain (although with the longer strings for storage you may need something like IPFS) so I would say that you shouldn't really need to use a secret if you make it decentralized. If you're using a server for storing state that sort of defeats the purpose of using blockchain to a certain degree.

**Comment by YalorMewn** [comment]
Hey @wip-abramson your mentor for the Blockternship is @Roots2D from #chainshot, the place to get in touch with him is [Slack](https://chainshotnodes.slack.com/join/shared_invite/enQtMzU3ODc5NTM3MTI3LTFlZTY1YzcwM2QzYWI0ODY2ZDczMmYzOTVlYWQwZjkyZDFlYzUxZWM4NDNlNjk3N2EyNGMwOGQ0ZTVkZjQyNjE)

**Comment by status-open-bounty** [comment]
Current balance: 0.0 ETH
Tokens: DAI: 115.00
Contract address: [0xf0c6aa3abfdbffdfd6c20dd4b209536ff1847b33](https://etherscan.io/address/0xf0c6aa3abfdbffdfd6c20dd4b209536ff1847b33)
![QR Code](https://openbounty.status.im/qr/Blockternship/projects/bounty/5/8b30bddea0e7f9825bb9b247727018fe6c7bd02d0461db4bf47cf2e93a871934/qr.png)
Network: Mainnet
To claim this bounty sign up at [https://openbounty.status.im](https://openbounty.status.im) and make sure to update your Ethereum address in `My Payment Details` so that the bounty is correctly allocated.
To fund it, send ETH or ERC20/ERC223 tokens to the contract address.

**Comment by YalorMewn** [comment]
Thanks for participating, don't forget to collect you bounty !
